### PR TITLE
enable strictPropertyInitialization in foundation

### PR DESCRIPTION
**Pull request by m4thieulavoie**
# Pull Request

## 📖 Description

This PR aims to re-enable `strictPropertyInitialization` in `fast-foundation`. It's still in draft mode because I'm not sure I'm going the right way. See questions in the comments below :)

### 🎫 Issues

https://github.com/microsoft/fast/issues/5338

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-foundation/src/accordion-item/accordion-item.ts [hunk 1]
```diff
@@ -56,12 +56,12 @@ export class AccordionItem extends FoundationElement {
      * HTML Attribute: id
      */
     @attr
-    public id: string;
+    public id: string | undefined;
```
Comment (m4thieulavoie) [inline packages/web-components/fast-foundation/src/accordion-item/accordion-item.ts:59]: For primitives, are we preferring
```suggestion
    public id: string | undefined;
```
or
```suggestion
    public id: string = "";
```

File: packages/web-components/fast-foundation/src/anchor/anchor.ts [hunk 2]
```diff
@@ -81,7 +81,7 @@ export class Anchor extends FoundationElement {
      * HTML Attribute: target
      */
     @attr
-    public target: "_self" | "_blank" | "_parent" | "_top";
+    public target: "_self" | "_blank" | "_parent" | "_top" | undefined;
```
Comment (m4thieulavoie) [inline packages/web-components/fast-foundation/src/anchor/anchor.ts:84]: Should I set the default value here? I _think_ it'd be `_self` ?

File: packages/web-components/fast-foundation/src/anchor/anchor.ts [hunk 3]
```diff
@@ -99,12 +99,12 @@ export class Anchor extends FoundationElement {
      * @internal
      */
     @observable
-    public defaultSlottedContent: HTMLElement[];
+    public defaultSlottedContent: HTMLElement[] | undefined;
```
Comment (m4thieulavoie) [inline packages/web-components/fast-foundation/src/anchor/anchor.ts:102]: Should I default to `[]` instead?

**Comment by m4thieulavoie** [comment]
@chrisdholt I could use some guidance for this one :pray:

**Comment by chrisdholt** [comment]
Here we prefer the references to undefined rather than an empty string value for strings, but as I found [here](https://github.com/microsoft/fast/issues/5320#issuecomment-1136456700) I don't think we'll be able to do it. I'm going to close this and the corresponding issue but ideally we can still have `noImplicitAny`.

**Comment by chrisdholt** [comment]
Thanks for putting in work on this @m4thieulavoie - sorry we couldn't make this one happen!

**Pull request closed without merging**